File: VB/Applications/WpfLog4NetVBNoDI/Extensions/ServicesExtension.vb
Imports System.Runtime.CompilerServices
Imports log4net.Appender.LogViewVB.Core
Imports LogViewerVB.Core
Imports Microsoft.Extensions.Configuration
Imports Microsoft.Extensions.DependencyInjection
Imports Microsoft.Extensions.Logging
Imports WpfLog4NetVBNoDI.DataStores

Public Module ServicesExtension

    <Extension>
    Public Function AddLog4NetNoDI(builder As ILoggingBuilder, config As IConfiguration) As ILoggingBuilder

        ' We need to use a shared instance of the DataStore to pass to the LogViewerControl
        builder.Services.AddSingleton(MainControlsDataStore.DataStore)

        ' call core Log4Net ServiceExtension initializer
        builder.AddLog4Net(config)

        Return builder

    End Function

    <Extension>
    Public Function AddLog4NetNoDI(builder As ILoggingBuilder, config As IConfiguration, configure As Action(Of DataStoreLoggerConfiguration)) As ILoggingBuilder

        builder.AddLog4NetNoDI(config)
        builder.Services.Configure(configure)
        Return builder

    End Function

End Module

File: CSharp/Applications/WinFormsLog4NetNoDI/Extensions/ServicesExtension.cs
using LogViewer.Core;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.Logging;
using Microsoft.Extensions.DependencyInjection;
using WinFormsLog4NetNoDI.DataStores;
using Log4Net.Appender.LogView.Core;

namespace WinFormsLog4NetNoDI.Extensions;

public static class ServicesExtension
{
    public static ILoggingBuilder AddLog4NetNoDI(this ILoggingBuilder builder, IConfiguration config)
    {
        // We need to use a shared instance of the DataStore to pass to the LogViewerControl
        builder.Services.AddSingleton(MainControlsDataStore.DataStore);

        // call core Log4Net ServiceExtension initializer
        builder.AddLog4Net(config);

        return builder;
    }

    public static ILoggingBuilder AddLog4NetNoDI(this ILoggingBuilder builder, IConfiguration config, Action<DataStoreLoggerConfiguration> configure)
    {
        builder.AddLog4NetNoDI(config);
        builder.Services.Configure(configure);
        return builder;
    }
}